libc/benchmarks/gpu/LibcGpuBenchmark.h

```diff
@@ -105,6 +109,56 @@ class Benchmark {
     return benchmark(options, func);
   }
 };
+
+// We want our random values to be approximately
+// |real value| <= 2^(max_exponent) * (1 + (random 52 bits) * 2^-52) <
+// 2^(max_exponent + 1)
+// The largest integer that can be stored in a double is 2^53
+static constexpr int MAX_EXPONENT = 52;
+
+static double get_rand_double() {
```

Comment:
We should probably template this because we'll want the same treatment for `sin` and `sinf` and `sinf16`.

Reply:
Since floats should have a different MAX_EXPONENT, how would you suggest we template this? Is there a nice way to switch the value of `MAX_EXPONENT`when using floats and doubles?

Reply:
`FPBits<T>::FRACTION_LEN`

Reply:
I forget the name of it, but it's standard `type_traits` stuff, see https://godbolt.org/z/4TGrvadYY.